File: 050_Pow-x-y/50.swift
class Solution {
    func myPow(_ x: Double, _ n: Int) -> Double {
        
        guard n != 0 else { return 1 }
        var answer = myPow(x, abs(n) / 2)
        
        answer = n % 2 == 0 ? answer * answer : answer * answer * x
        return n > 0 ? answer : 1 / answer
    }
}

File: 0050_Pow-x-y/50.cpp
#include <iostream>>
#include <stdlib.h>

using namespace std;


class Solution {
public:
    double myPow(double x, int n) {
        if (n == 0) {
            return 1;
        }
        auto answer = myPow(x, abs(n) / 2);

        answer = (n % 2 == 0) ? answer * answer : answer * answer * x;

        return n > 0 ? answer : 1 / answer;
    }
};